src/python/CRABInterface/DataWorkflow.py

```diff
@@ -168,12 +167,10 @@ def submit(self, workflow, activity, jobtype, jobsw, jobarch, use_parent, second
             collector = backend_urls['htcondorPool']
 
         try:
-            requestname = '%s:%s_%s' % (timestamp, userhn, workflow)
+            taskname = '%s:%s_%s' % (timestamp, userhn, workflow)
             schedd_name = self.chooseScheduler(scheddname, backend_urls).split(":")[0]
-            if len(requestname) > RX_WORKFLOW_LEN:
-              raise InvalidParameter("Taskname exceed maximum allowed lenght, please review parameter requestName in your config file")
```

Comment:
Mmhh... I think that if the check here is just to make sure that the length of the taskname is not greater than the column size in the Oracle Task DB, then it shouldn't be needed. Oracle itself will not allow the writing. What we should instead do is to recognize the error returned by Oracle (I am not sure if by catching an exception or what) and report back the error to the client with a clear message. And this would cover ANY writing/reading error to/from the Oracle DB. 
I just tested what happens if I submit a task that tries to write a taskname with length > 255 into the database (I hacked my code to not validate the taskname beforehand). This is what you get when you submit:
```
$ crab submit
Will use CRAB configuration file crabConfig.py
Importing CMSSW configuration pset_tutorial_analysis.py
Finished importing CMSSW configuration pset_tutorial_analysis.py
Sending the request to the server
The server answered with an error.
Server answered with: Execution error
Error Id: 180bb96143fe0146be42acd62ba07e79
Log file is /afs/cern.ch/work/a/atanasi/crab3/CMSSW_7_3_5_patch2/src/crab_test_loooooong_taskname_245_eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee/crab.log
```
And this is what appears in the crabserver logs:
```
[16/Nov/2015:10:26:53]  RESTSQL:GVfuyzDeSWdo ENTER atanasi@devdb11 CRABInterface.RESTBaseAPI.RESTBaseAPI (PUT dev workflow) inuse=0 idle=1
[16/Nov/2015:10:26:53]  RESTSQL:GVfuyzDeSWdo (previously SJjiYdrxAGVM) connected, client: 11.2.0.3.0, server: 11.2.0.4.0, stmtcache: 50
[16/Nov/2015:10:26:53]  RESTSQL:GVfuyzDeSWdo ping
[16/Nov/2015:10:26:53]  RESTSQL:GVfuyzDeSWdo check [select sysdate from dual]
[16/Nov/2015:10:26:53]  RESTSQL:GVfuyzDeSWdo connection established
[16/Nov/2015:10:26:54]  SERVER DATABASE ERROR 500/403 Execution error cx_Oracle.DatabaseError 180bb96143fe0146be42acd62ba07e79 [instance: dev] (ORA-12899: value too large for column "ATANASI"."TASKS"."TM_TASKNAME" (actual: 267, maximum: 255)); last statement: INSERT INTO tasks (               tm_taskname, tm_activity, panda_jobset_id, tm_task_status, tm_start_time, tm_task_failure, tm_job_sw,               tm_job_arch, tm_input_dataset, tm_primary_dataset, tm_nonvalid_input_dataset, tm_use_parent, tm_secondary_input_dataset, tm_site_whitelist, tm_site_blacklist,               tm_split_algo, tm_split_args, tm_totalunits, tm_user_sandbox, tm_cache_url, tm_username, tm_user_dn,               tm_user_vo, tm_user_role, tm_user_group, tm_publish_name, tm_publish_groupname, tm_asyncdest, tm_dbs_url, tm_publish_dbs_url,               tm_publication, tm_outfiles, tm_tfile_outfiles, tm_edm_outfiles, tm_job_type, tm_generator, tm_arguments,               panda_resubmitted_jobs, tm_save_logs, tm_user_infiles, tm_maxjobruntime, tm_numcores, tm_maxmemory, tm_priority,               tm_scriptexe, tm_scriptargs, tm_extrajdl, tm_asourl, tm_events_per_lumi, tm_collector, tm_schedd, tm_dry_run,               tm_user_files, tm_transfer_outputs, tm_output_lfn, tm_ignore_locality, tm_fail_limit, tm_one_event_mode)               VALUES (:<**>task_name, :task_activity, :jobset_id, upper(:task_status), SYS_EXTRACT_UTC(SYSTIMESTAMP), :task_failure, :job_sw,               :job_arch, :input_dataset, :primary_dataset, :nonvalid_data, :use_parent, :secondary_dataset, :site_whitelist, :site_blacklist,               :split_algo, :split_args, :total_units, :user_sandbox, :cache_url, :username, :user_dn,               :user_vo, :user_role, :user_group, :publish_name, :publish_groupname, :asyncdest, :dbs_url, :publish_dbs_url,               :publication, :outfiles, :tfile_outfiles, :edm_outfiles, :job_type, :generator, :arguments,               :resubmitted_jobs, :save_logs, :user_infiles, :maxjobruntime, :numcores, :maxmemory, :priority,               :scriptexe, :scriptargs, :extrajdl, :asourl, :events_per_lumi, :collector, :schedd_name, :dry_run,               :user_files, :transfer_outputs, :output_lfn, :ignore_locality, :fail_limit, :one_event_mode); binds: ([{'maxjobruntime': 1250, 'extrajdl': '[]', 'events_per_lumi': None, 'user_dn': u'/DC=ch/DC=cern/OU=Organic Units/OU=Users/CN=atanasi/CN=710186/CN=Andres Jorge Tanasijczuk', 'user_infiles': '[]', 'user_files': '[]', 'job_type': 'Analysis', 'ignore_locality': 'F', 'scriptexe': None, 'job_sw': 'CMSSW_7_3_5_patch2', 'asyncdest': 'T2_US_Nebraska', 'collector': 'cmssrv221.fnal.gov,vocms099.cern.ch', 'task_failure': '', 'numcores': 1, 'publication': 'F', 'generator': 'pythia', 'maxmemory': 2000, 'publish_dbs_url': 'https://cmsweb.cern.ch/dbs/prod/phys03/DBSWriter', 'priority': 10, 'dbs_url': 'https://cmsweb.cern.ch/dbs/prod/global/DBSReader', 'arguments': '{}', 'resubmitted_jobs': '[]', 'schedd_name': 'crab3test-2@vocms095.cern.ch', 'primary_dataset': 'DoubleMuParked', 'transfer_outputs': 'F', 'username': 'atanasi', 'fail_limit': None, 'site_blacklist': '[]', 'secondary_dataset': None, 'publish_name': 'test_loooooong_taskname_245-00000000000000000000000000000000', 'scriptargs': '[]', 'output_lfn': '/store/user/atanasi/trash', 'tfile_outfiles': '[]', 'one_event_mode': 'F', 'user_role': None, 'input_dataset': '/DoubleMuParked/Run2012B-22Jan2013-v1/AOD', 'total_units': 1, 'split_args': "{'splitOnRun': False, 'halt_job_on_file_boundaries': False, 'lumis': [], 'runs': [], 'lumis_per_job': 1}", 'user_group': None, 'split_algo': 'LumiBased', 'edm_outfiles': "['output1.root']", 'publish_groupname': 'F', 'cache_url': 'https://osmatanasi2.cern.ch/crabcache', 'jobset_id': None, 'nonvalid_data': 'F', 'dry_run': 'F', 'site_whitelist': '[]', 'task_activity': 'analysistest', 'user_sandbox': 'hola.tar.gz', 'asourl': 'https://cmsweb-testbed.cern.ch/couchdb', 'user_vo': 'cms', 'task_name': '151116_092653:atanasi_crab_test_loooooong_taskname_245_eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee', 'outfiles': '[]', 'task_status': 'NEW', 'job_arch': 'slc6_amd64_gcc491', 'use_parent': 0, 'save_logs': 'F'}],), {}; offset: 1077
[16/Nov/2015:10:26:54]    Traceback (most recent call last):
[16/Nov/2015:10:26:54]      File "/afs/cern.ch/work/a/atanasi/private/github/repos/WMCore/src/python/WMCore/REST/Server.py", line 1728, in dbapi_wrapper
[16/Nov/2015:10:26:54]        return handler(*xargs, **xkwargs)
[16/Nov/2015:10:26:54]      File "/afs/cern.ch/work/a/atanasi/private/github/repos/CRABServer/src/python/CRABInterface/RESTUserWorkflow.py", line 534, in put
[16/Nov/2015:10:26:54]        scriptexe=scriptexe, scriptargs=scriptargs, scheddname=scheddname, extrajdl=extrajdl, collector=collector, dryrun=dryrun)
[16/Nov/2015:10:26:54]  RESTSQL:GVfuyzDeSWdo disconnecting
[16/Nov/2015:10:26:54]      File "/afs/cern.ch/work/a/atanasi/private/github/repos/CRABServer/src/python/CRABInterface/DataUserWorkflow.py", line 129, in submit
[16/Nov/2015:10:26:54]        return self.workflow.submit(*args, **kwargs)
[16/Nov/2015:10:26:54]      File "/afs/cern.ch/work/a/atanasi/private/github/repos/CRABServer/src/python/CRABInterface/Utils.py", line 144, in wrapped_func
[16/Nov/2015:10:26:54]        return func(*args, **kwargs)
[16/Nov/2015:10:26:54]      File "/afs/cern.ch/work/a/atanasi/private/github/repos/CRABServer/src/python/CRABInterface/DataWorkflow.py", line 255, in submit
[16/Nov/2015:10:26:54]        one_event_mode   = ['T' if oneEventMode else 'F']
[16/Nov/2015:10:26:54]      File "/afs/cern.ch/work/a/atanasi/private/github/repos/WMCore/src/python/WMCore/REST/Server.py", line 2162, in modify
[16/Nov/2015:10:26:54]        c, _ = self.executemany(sql, kwbinds, *binds)
[16/Nov/2015:10:26:54]      File "/afs/cern.ch/work/a/atanasi/private/github/repos/WMCore/src/python/WMCore/REST/Server.py", line 2071, in executemany
[16/Nov/2015:10:26:54]        return c, c.executemany(None, *binds, **kwbinds)
[16/Nov/2015:10:26:54]    DatabaseError: ORA-12899: value too large for column "ATANASI"."TASKS"."TM_TASKNAME" (actual: 267, maximum: 255)
[16/Nov/2015:10:26:54] osmatanasi2 188.184.69.232 "PUT /crabserver/dev/workflow HTTP/1.1" 500 Internal Server Error [data: 7191 in 726 out 2104350 us ] [auth: OK "/DC=ch/DC=cern/OU=Organic Units/OU=Users/CN=atanasi/CN=710186/CN=Andres Jorge Tanasijczuk" "" ] [ref: "" "CRABClient/development" ]
[16/Nov/2015:10:26:54]  RESTSQL:GVfuyzDeSWdo DISCONNECTED atanasi@devdb11 timeout=300 inuse=0 idle=0
``` 

If what you want to check here instead is that the taskname length is not larger than N with N < 255, then yes, we can keep it. But why would you restrict taskname to be shorted that N < 255? One reason is what I described in the PR, that filenames built out from the taskname plus something (like the TW log file copy that is uploaded to the crabcache for use by the CRAB Monitor) could then exceed 255 chars. But lets first decide whether limiting the taskname to N < 255 (e.g. 225) is the solution we want to take, at least momentarily, to solve this problem.